packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -1019,7 +1019,10 @@ class FlutterVmService {
         onError: (Object? error, StackTrace stackTrace) {
           if (error is vm_service.SentinelException ||
             error == null ||
-            (error is vm_service.RPCError && error.code == RPCErrorCodes.kServiceDisappeared)) {
+            error is vm_service.RPCError &&
+              (error.code == RPCErrorCodes.kServiceDisappeared ||
+                error.code == RPCErrorCodes.kInternalError &&
+                error.message.contains('Sentinel kind: Collected'))) {
```

Comment:
@bkonyi chrome proxy service throws a sentinel exception on getIsolate for non-existing isolate id, as required by the VmServiceInterface, but the VmServerConnection translates it into an RPC error with RPCError.kInternalError as code, and the sentinel is added to the message as a string, so I have to check for the string contents here. Is there a better way to handle this error? Should we change the VMServer connection to throw something different, or have a set of error codes in package:vm_service we all agree on and can check for here?

Reply:
`service.getIsolate(...)` should be throwing a `SentinelException` in this case, so there's likely something wrong with the encoding on the DWDS side of things. What's the JSON response you're seeing here? It should look something like:

```json
{
  "id": "123",
  "result": {
    "type": "Sentinel",
    "kind": "Collected",
    "valueAsString": "..."
  }
}
```

Reply:
Chrome proxy service throws a sentinel exception, but the vm server connection we use to on top of it converts it to RPC error with code of `RPCError.kInternalError` here: https://github.com/flutter/flutter/blob/961df985faa4db9eff0245b3faa066fd0278cee9/packages/flutter_tools/lib/src/vmservice.dart#L1025

Should this code be changed then?

Reply:
I will file an issue in the SDK repo for this, in the meanwhile the current PR should fix the flake.

Reply:
Filed an issue (i am worried about all the other cases where we miss error checks due to this problem): https://github.com/dart-lang/sdk/issues/51752